### DISCUSSION TITLE
Generate Report Designer tool,  (View Oxetta)

### DISCUSSION THREAD

**Discussion by esbvaGames** [utelle/wxpdfdoc#87] [Ideas]
Is there any possibility of creating a tool similar to "Oxetta", to generate reports from databases, based on XRC, html, or XML resources?

I will leave this link, although I did not manage to join the DLL well to my wxWidget App !!!

http://www.oxetta.com/report/doc/start.html

**Comment by utelle** [comment]
> Is there any possibility of creating a tool similar to "Oxetta", to generate reports from databases, based on XRC, html, or XML resources?

Sure. In principle, it is possible to develop such a report designer and such a report generator. However, that is certainly not a trivial task.

What I have implemented in one of my projects to generate PDF reports is the following procedure:

1. Retrieve data from database and generate XML containing the retrieved data
2. Convert the data XML via XSLT into a XML report (using a simple XML dialect describing the report - pages, tables and so on)
3. Interpret the XML report and generate PDF output from it using wxPdfDocument

This worked very well and allowed to generate rather complex multi-page reports.

> I will leave this link, although I did not manage to join the DLL well to my wxWidget App !!!
> 
> http://www.oxetta.com/report/doc/start.html

Well, Oxetta is closed source, developed with Visual C++ 2010, and last but not least rather dated (no updates after 2016, that is, almost 7 years ago). But what's even more important: the website shows no name of the developer(s), and no option to contact him/them in case of bugs. Besides that the Oxetta website is hosted in Russia. Please draw your own conclusions ...
